src/EditorFeatures/Core/FindReferences/FindReferencesCommandHandler.cs

```diff
@@ -127,14 +127,18 @@ private IStreamingFindUsagesPresenter GetStreamingPresenter()
                     // the context object that the FAR service will push results into.
                     var context = presenter.StartSearch(
                         EditorFeaturesResources.Find_References, alwaysShowDeclarations: false);
-                    await findUsagesService.FindReferencesAsync(document, caretPosition, context).ConfigureAwait(false);
-
-                    // Note: we don't need to put this in a finally.  The only time we might not hit
-                    // this is if cancellation or another error gets thrown.  In the former case,
-                    // that means that a new search has started.  We don't care about telling the
-                    // context it has completed.  In the latter case somethign wrong has happened
-                    // and we don't want to run any more code code in this particular context.
-                    await context.OnCompletedAsync().ConfigureAwait(false);
+
+                    using (Logger.LogBlock(FunctionId.CommandHandler_FindAllReference, context.CancellationToken))
```

Comment:
@CyrusNajmabadi If I wrap .StartSearch, I won't be able to use the CancellationToken from the context it produces. In that case, where do I get one? (Maybe this should be obvious).